src/keboola_mcp_server/tools/components/utils.py

```diff
@@ -416,6 +418,18 @@ async def set_cfg_update_metadata(
 # ============================================================================
 
 
+def get_nested(obj: Mapping[str, Any] | None, key: str, *, default: T = None) -> T | None:
```

Comment:
nitpick: maybe it should be `default: T | None = None`. Also, we can't guarantee that the return type will be `T` but that's fine